folium/folium.py

```diff
@@ -583,20 +578,44 @@ def geo_json(self, geo_path=None, geo_str=None, data_out='data.json',
             color_domain = [data_min+i*(data_max-data_min)*1./nb_class
                             for i in range(1+nb_class)]
         else:
-            color_domain = [-1, 1]
+            color_domain = None
+
+        if color_domain and key_on:
+            key_on = key_on[8:] if key_on.startswith('feature.') else key_on
+            color_range = color_brewer(fill_color, n=len(color_domain))
+            get_by_key = lambda obj,key : (obj.get(key,None) if len(key.split('.'))<=1
+                                           else get_by_key(obj.get(key.split('.')[0],None),
+                                                           '.'.join(key.split('.')[1:])))
+            color_scale_fun = lambda x: color_range[len([u for u in color_domain
+                                                         if u<=color_data[get_by_key(x,key_on)]])]
+        else:
+            color_scale_fun = lambda x: fill_color
```

Comment:
If my understanding is correct the choropleth would be virtually this function, but renamed and with a few modifications.  Correct?

Since we are killing the `geo_json()` anyways, maybe we can just replace by the new `choropleth()` in the next PR.